CMakeLists.txt

```diff
@@ -244,17 +284,21 @@ configure_file(
 
 if(JAEGERTRACING_BUILD_EXAMPLES)
   add_executable(app examples/App.cpp)
-  target_link_libraries(app PUBLIC ${JAEGERTRACING_LIB})
+  target_link_libraries(app PUBLIC jaegertracing)
 endif()
 
 if(BUILD_TESTING)
+
+  set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
```

Comment:
Again, unsure if this is ideal. @ruslo should have some ideas here too.

Reply:
It depends on your goal/approach. You can use `CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS` to export everything or use [GenerateExportHeader](https://cmake.org/cmake/help/latest/module/GenerateExportHeader.html) module and mark only particular classes/methods with `*_EXPORT`:
* https://github.com/cgold-examples/fruits/blob/081f76a970d3ebbc8d316ad8ef52994b8b4cad56/lib/fruits/rosaceae/Plum.hpp#L9
* https://github.com/cgold-examples/fruits/blob/081f76a970d3ebbc8d316ad8ef52994b8b4cad56/lib/fruits/rosaceae/Pear.hpp#L11

However I think if you decide to export everything then `set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)` should be set once in top level `CMakeLists.txt` and there is no need to unset it, `set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS OFF)` can be removed.

Reply:
So I think I had wanted to do the export header thing, just never got around to that. I was just wondering on your opinion about `set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)`. It seems like something for `toolchain.cmake`. I know you do not believe `set(CMAKE_CXX_STANDARD 11)` is a good idea in CMakeLists.txt. I wonder if this could be a similar dangerous option.

Reply:
I had a dffierent approach at the beginning where I exported the symbols one by one. I found that fastidious and ended with a huge list. That's why I adopted the approach of exporting all symbols.

Reply:
I'm thinking of going the route that jaeger-client-java did. Essentially, why export any of the symbols other than perhaps the config? The OpenTracing API is the only public API we recommend to users.

Reply:
I think that this can be a good idea. However, this means that won't be able to test the Jaeger code directly but we'll have to go through the OpenTracing API.